osutil/disks/disks_linux.go

```diff
@@ -251,6 +251,24 @@ func DiskFromMountPoint(mountpoint string, opts *Options) (Disk, error) {
 	return diskFromMountPoint(mountpoint, opts)
 }
 
+// DiskFromPartitionDeviceNode finds a matching Disk that the specified
+// partition node resides on.
+func DiskFromPartitionDeviceNode(node string) (Disk, error) {
+	// TODO: support options such as IsDecryptedDevice
+	return diskFromPartitionDeviceNode(node)
+}
+
+var diskFromPartitionDeviceNode = func(node string) (Disk, error) {
+	// get the udev properties for this device node
+	props, err := udevPropertiesForName(node)
+	if err != nil {
+		return nil, err
+	}
+
+	errId := fmt.Sprintf("partition device node %s", node)
+	return diskFromPartUdevPropsAndOpts(errId, props, nil)
```

Comment:
I think the errID should be really added by the caller, as there is more context available there:
```
disk, err := diskFromPartUdevPropsAndOpts(props, nil)
if err != nil {
    return fmt.Errorf("cannot find disk from partition device node %s: %v", node, err)
}
return disk, nil
```

Reply:
done